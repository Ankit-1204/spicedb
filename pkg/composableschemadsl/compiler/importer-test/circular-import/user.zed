from .subjects import persona

definition user {}
