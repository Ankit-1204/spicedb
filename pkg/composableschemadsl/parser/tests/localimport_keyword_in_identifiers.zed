from .path.to.user import user, caveat

definition resource {
    relation user: user
    relation persona: persona
    permission view = user + persona
}
