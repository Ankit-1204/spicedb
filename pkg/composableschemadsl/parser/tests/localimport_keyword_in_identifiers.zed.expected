NodeTypeFile
  end-rune = 154
  input-source = local imports with keyword in identifiers test
  start-rune = 0
  child-node =>
    NodeTypeImport
      end-rune = 30
      input-source = local imports with keyword in identifiers test
      start-rune = 0
      imported-definition =>
        NodeTypeIdentifier
          end-rune = 29
          identifier-value = user
          input-source = local imports with keyword in identifiers test
          start-rune = 26
        NodeTypeIdentifier
          end-rune = 30
          input-source = local imports with keyword in identifiers test
          start-rune = 32
          child-node =>
            NodeTypeError
              end-rune = 30
              error-message = Expected identifier, found token TokenTypeKeyword
              error-source = caveat
              input-source = local imports with keyword in identifiers test
              start-rune = 32
      path-segment =>
        NodeTypeIdentifier
          end-rune = 9
          identifier-value = path
          input-source = local imports with keyword in identifiers test
          start-rune = 6
        NodeTypeIdentifier
          end-rune = 12
          identifier-value = to
          input-source = local imports with keyword in identifiers test
          start-rune = 11
        NodeTypeIdentifier
          end-rune = 17
          identifier-value = user
          input-source = local imports with keyword in identifiers test
          start-rune = 14
    NodeTypeCaveatDefinition
      end-rune = 37
      input-source = local imports with keyword in identifiers test
      start-rune = 32
      child-node =>
        NodeTypeError
          end-rune = 37
          error-message = Expected identifier, found token TokenTypeSyntheticSemicolon
          error-source = 

          input-source = local imports with keyword in identifiers test
          start-rune = 38
    NodeTypeDefinition
      definition-name = resource
      end-rune = 153
      input-source = local imports with keyword in identifiers test
      start-rune = 40
      child-node =>
        NodeTypeRelation
          end-rune = 84
          input-source = local imports with keyword in identifiers test
          relation-name = user
          start-rune = 66
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 84
              input-source = local imports with keyword in identifiers test
              start-rune = 81
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 84
                  input-source = local imports with keyword in identifiers test
                  start-rune = 81
                  type-name = user
        NodeTypeRelation
          end-rune = 114
          input-source = local imports with keyword in identifiers test
          relation-name = persona
          start-rune = 90
          allowed-types =>
            NodeTypeTypeReference
              end-rune = 114
              input-source = local imports with keyword in identifiers test
              start-rune = 108
              type-ref-type =>
                NodeTypeSpecificTypeReference
                  end-rune = 114
                  input-source = local imports with keyword in identifiers test
                  start-rune = 108
                  type-name = persona
        NodeTypePermission
          end-rune = 151
          input-source = local imports with keyword in identifiers test
          relation-name = view
          start-rune = 120
          compute-expression =>
            NodeTypeUnionExpression
              end-rune = 151
              input-source = local imports with keyword in identifiers test
              start-rune = 138
              left-expr =>
                NodeTypeIdentifier
                  end-rune = 141
                  identifier-value = user
                  input-source = local imports with keyword in identifiers test
                  start-rune = 138
              right-expr =>
                NodeTypeIdentifier
                  end-rune = 151
                  identifier-value = persona
                  input-source = local imports with keyword in identifiers test
                  start-rune = 145